### PR TITLE
added test to check parameter existing-cluster=true

### PR DESCRIPTION
**Pull request by tikitavi**

### CODE REVIEW

File: juju/scaleio/deploy-scaleio-cluster.sh [hunk 1]
```diff
@@ -32,15 +31,15 @@ echo "Machine created: $m3"
...
+$my_dir/../scaleio-openstack/fix_scini_problems.sh $m1 $m2 $m3
 
 # ---------------------------------------- pre-deployment stage end
 
 # change bundles' variables
 sed -i -e "s/%JUJU_REPO%/$JUJU_REPO/m" $BUNDLE
 
 # script needs to change directory to local charms repository
-cd juju-scaleio
+cd ~/juju-scaleio
```
Comment (Andrey-mp) [inline juju/scaleio/deploy-scaleio-cluster.sh]: this will not work in Jenkins

Comment (Andrey-mp) [inline juju/scaleio/deploy-scaleio-cluster.sh]: just revert it

File: juju/fuel/deploy-fuel-existing-cluster.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+my_file="$(readlink -e "$0")"
+my_dir="$(dirname $my_file)"
+
+source $my_dir/functions
+
+# Deploy bundle
+$my_dir/../scaleio/deploy-scaleio-cluster.sh
+# TODO: add deployed machines to array automaticly
+machines=(1 2 3)
```
Comment (Andrey-mp) [inline juju/fuel/deploy-fuel-existing-cluster.sh]: do you really need these machines in array?
you only need gateway params - ip/user/password

File: juju/fuel/deploy-fuel-existing-cluster.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+# provision machines
+provision_machines 3 4 5 6
```
Comment (Andrey-mp) [inline juju/fuel/deploy-fuel-existing-cluster.sh]: and these indexes can start from 0

File: juju/fuel/functions [hunk 4]
```diff
@@ -531,3 +532,79 @@ function check_scaleio_not_installed {
...
+
+  cluster_mdm_ips=(`juju ssh $mdm "scli --query_cluster --approve_certificate" 2>/dev/null | awk '/Management IPs/{print$5}' | sed 's/,//g'`)
+
+  for node in $sdc_nodes ; do
+    sdc=${machines[$node]}
+    #check emc-packages"
+    if output=`juju ssh $sdc "dpkg -l emc*" 2>/dev/null | grep emc | grep -v 'scaleio-sdc'` ; then
+      echo "ERROR: There is scaleio packages on machine $sdc besides scaleio-sdc."
+      echo $output
+      return 1
```
Comment (Andrey-mp) [inline juju/fuel/functions]: why return ?

File: juju/fuel/functions [hunk 5]
```diff
@@ -531,3 +532,79 @@ function check_scaleio_not_installed {
...
+function check_existing_cluster {
+  mdm=`get_master_mdm`
+  sdc_nodes=`echo $1 | sed 's/,/ /g'`
+
+  cluster_mdm_ips=(`juju ssh $mdm "scli --query_cluster --approve_certificate" 2>/dev/null | awk '/Management IPs/{print$5}' | sed 's/,//g'`)
+
+  for node in $sdc_nodes ; do
+    sdc=${machines[$node]}
+    #check emc-packages"
+    if output=`juju ssh $sdc "dpkg -l emc*" 2>/dev/null | grep emc | grep -v 'scaleio-sdc'` ; then
```
Comment (Andrey-mp) [inline juju/fuel/functions]: 'grep emc' - are there any other packages present with 'emc' substring?

File: juju/fuel/functions [hunk 6]
```diff
@@ -531,3 +532,79 @@ function check_scaleio_not_installed {
...
+      echo "ERROR: There is scaleio packages on machine $sdc besides scaleio-sdc."
+      echo $output
+      return 1
+    else
+      echo "INFO: Success. There is no scaleio packages on machine $sdc besides scaleio-sdc."
+    fi
+
+    sdc_mdm_ips=(`juju ssh $sdc "sudo /bin/emc/scaleio/drv_cfg --query_mdms" 2>/dev/null | awk '/MDM-ID/{print$10}' | sed 's/\[0\]-//g'`)
```
Comment (Andrey-mp) [inline juju/fuel/functions]: awk: insert space between print and $10

File: juju/fuel/functions (hunk 4 above)
Comment (tikitavi) [inline juju/fuel/functions]: There should not be other emc packages on additional nodes. If they are, something went wrong. No need to continue test.

File: juju/fuel/deploy-fuel-existing-cluster.sh [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+# Deploy bundle
+$my_dir/../scaleio/deploy-scaleio-cluster.sh
+
+# provision machines
+provision_machines 0 1 2 3
+# prepare fuel master
+prepare_fuel_master 0
+
+gateway_ip=(`juju status scaleio-gw | grep public-address | awk '{print $2}'`)
```
Comment (Andrey-mp) [inline juju/fuel/deploy-fuel-existing-cluster.sh]: why it is array?

File: juju/fuel/functions (hunk 4 above)
Comment (Andrey-mp) [inline juju/fuel/functions]: what about other nodes?

File: juju/fuel/functions [hunk 8]
```diff
@@ -531,3 +522,91 @@ function check_scaleio_not_installed {
...
+  cluster_mdm_ips=(`juju ssh $mdm "scli --query_cluster --approve_certificate" 2>/dev/null | awk '/Management IPs/{print$5}' | sed 's/,//g'`)
+
+  error_nodes=()
+  for node in $sdc_nodes ; do
+    sdc=${machines[$node]}
+    #check emc-packages"
+    if output=`juju ssh $sdc "dpkg -l emc*" 2>/dev/null | grep emc | grep -v 'scaleio-sdc'` ; then
+      echo "ERROR: There is scaleio packages on machine $sdc besides scaleio-sdc."
+      echo $output
+      error_nodes+=$sdc
```
Comment (Andrey-mp) [inline juju/fuel/functions]: error...

**Pull request closed without merging**